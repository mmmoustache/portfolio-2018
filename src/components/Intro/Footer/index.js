import styled from "styled-components";
import { convertRemToPixels as rem } from '../../../utils/utils';

const offset = rem('40px');

export default styled.div`
  bottom: ${offset};
  display: flex;
  position: absolute;
  right: ${offset};
  z-index: 2;
`;
