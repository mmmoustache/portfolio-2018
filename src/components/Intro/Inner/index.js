import styled from "styled-components";
import { convertRemToPixels as rem } from '../../../utils/utils';
import { Colours } from '../../../utils/colours';
import { Layouts } from '../../../utils/layouts';

export default styled.div`
  color: ${Colours.white};
  margin: 0 auto;
  max-width: ${Layouts.constraint};
  padding: ${rem('20px')};
  position: relative;
  top: 50%;
  transform: translateY(-50%);
  z-index: 2;
`;

