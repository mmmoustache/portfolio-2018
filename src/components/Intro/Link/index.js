import styled from "styled-components";
import { Timings, Durations } from '../../../utils/transitions';
import { convertRemToPixels as rem } from '../../../utils/utils';

const LinkSize = rem('40px');
const LinkOffset = rem('10px');
const LinkSpacing = rem('20px');

const Link = styled.a`
  display: block;
  height: ${LinkSize};
  margin-left: ${LinkSpacing};
  opacity: 0;
  position: relative;
  text-decoration: none;
  transform: translateY(${LinkOffset});
  transition: opacity ${Durations.fast} ${Timings.default}, transform ${Durations.fast} ${Timings.default};
  width: ${LinkSize};

  &:hover {
    opacity: 0.7;
    transition-delay: 0s;
  }
  
  ${
    props => props.loaded && `
      opacity: 1;
      transform: translateY(0);
    `
  }

  &::before {
    background-position: center;
    background-repeat: no-repeat;
    background-size: contain;
    content: '';
    display: block;
    height: ${LinkSize};
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: ${LinkSize};
  }
`;

export const Github = Link.extend`
  transition-delay: ${1*0.125}s;

  &::before {
    background-image: url('./github.svg');
  }
`;

export const LinkedIn = Link.extend`
  transition-delay: ${2*0.125}s;

  &::before {
    background-image: url('./linkedin.svg');
  }
`;
