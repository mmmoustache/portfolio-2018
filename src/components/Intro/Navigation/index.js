import styled from "styled-components";
import { Colours } from '../../../utils/colours';
import { Timings, Durations } from '../../../utils/transitions';
import { Border as BorderSettings } from '../../../utils/layouts';
import { convertRemToPixels as rem } from '../../../utils/utils';

const NavOffset = rem('40px');

export const Navigation = styled.div`
  position: absolute;
  right: ${NavOffset};
  top: ${NavOffset};
  z-index: 2;
`;

export const NavigationItem = styled.a`
  color: ${Colours.white};
  display: inline-block;
  font-weight: bold;
  opacity: 0;
  margin: 10px 20px;
  position: relative;
  text-decoration: none;
  transform: translateY(-10px);
  transition: opacity ${Durations.fast} ${Timings.default}, transform ${Durations.fast} ${Timings.default};

  ${
    props => props.loaded && `
      opacity: 1;
      transform: translateY(0);
    `
  }

  ${
    props => props.index && `
      transition-delay: ${props.index*0.125}s;
    `
  }

  &::before {
    background: ${Colours.primary};
    bottom: -${rem('5px')};
    content: '';
    display: block;
    height: ${BorderSettings.size};
    left: 0;
    position: absolute;
    transition: width ${Durations.ultrafast} ${Timings.default};
    width: 0;
    z-index: 1;
  }

  &:focus,
  &:hover {
    &::before {
      width: 100%;
    }
  }
`;
