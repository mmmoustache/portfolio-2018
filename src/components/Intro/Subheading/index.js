import styled from "styled-components";
import { Timings } from '../../../utils/transitions';
import { convertRemToPixels as rem } from '../../../utils/utils';
import { Breakpoints } from '../../../utils/breakpoints';

export default styled.div`
  height: auto;
  margin: 0 0 ${rem('15px')};
  overflow: hidden;
  transition: height 0.375s ${Timings.default};

  @media ${Breakpoints.medium} {
    margin: 0;

    ${
      props => props.loaded ? `height: ${rem('40px')};` : 'height: 0px;'
    }
  }
`;
