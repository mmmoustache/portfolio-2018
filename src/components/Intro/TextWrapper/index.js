import styled from "styled-components";
import { Colours } from '../../../utils/colours';

export default styled.div`
  position: relative;

  &::before {
    background: ${Colours.black};
    left: -100%;
    position: absolute;
    height: 80%;
    width: 100%;
    top: 0;
  }

  ${
    props => props.loaded && `
      & > * {
        opacity: 1;
        top: 0;
      }
    `
  }
`;
