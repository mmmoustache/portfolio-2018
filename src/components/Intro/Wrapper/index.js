import styled from "styled-components";
import { Colours } from '../../../utils/colours';

export default styled.div`
  background: ${Colours.secondary};
  height: 100vh;
  overflow: hidden;
  position: relative;
  width: 100%;
`;
