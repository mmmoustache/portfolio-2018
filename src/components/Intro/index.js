import React from 'react';
import PropTypes from 'prop-types';
import H1 from '../H1';
import H3 from '../H3';
import Wrapper from './Wrapper';
import { Navigation, NavigationItem as Link } from './Navigation';
import Footer from './Footer';
import TextWrapper from './TextWrapper';
import Subheading from './Subheading';
import Inner from './Inner';
import { LinkedIn, Github } from './Link';
import { Wrapper as BorderWrapper, Border1 as B1, Border2 as B2, Border3 as B3, Border4 as B4 } from "../Border";

const Intro = ({ loaded, navigation, linkedIn, gitHub }) => (
  <Wrapper loaded={loaded}>
    <BorderWrapper showOnMobile>
      <B1 isVisible={loaded} altColour />
      <B2 isVisible={loaded} altColour />
      <B3 isVisible={loaded} altColour />
      <B4 isVisible={loaded} altColour />
    </BorderWrapper>
    <Navigation>
      { navigation && navigation.map(e => <Link key={e.id} href={e.href} download={e.id === 3} index={e.id} loaded={loaded}>{e.title}</Link>)}
    </Navigation>
    <Footer>
      <LinkedIn loaded={loaded} href={linkedIn} title="View my LinkedIn Profile" target="_blank" rel="noopener noreferrer" />
      <Github loaded={loaded} href={gitHub} title="View my GitHub Profile" target="_blank" rel="noopener noreferrer" />
    </Footer>
    <Inner>
      <TextWrapper loaded={loaded}>
        <Subheading loaded={loaded}>
          <H3 noMargin>Front End Web Developer</H3>
        </Subheading>
        <H1>jack coventry</H1>
      </TextWrapper>
    </Inner>
  </Wrapper>
);

Intro.propTypes = {
  loaded: PropTypes.bool,
  navigation: PropTypes.array,
  linkedIn: PropTypes.string,
  gitHub: PropTypes.string,
};

Intro.defaultProps = {
  loaded: false,
  navigation: undefined,
  linkedIn: undefined,
  gitHub: undefined,
};

export default Intro;
