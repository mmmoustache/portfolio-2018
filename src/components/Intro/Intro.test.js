import React from 'react';
import renderer from 'react-test-renderer';
import Intro from "../Intro";

describe('Component: Intro', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <Intro loaded navigation={[]} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
