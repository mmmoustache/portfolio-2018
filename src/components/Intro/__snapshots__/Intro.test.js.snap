// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Intro Should render successfully 1`] = `
<div
  className="sc-htpNat iPkDTG"
>
  <div
    className="sc-gqjmRU zceRv"
  >
    <div
      className="sc-jTzLTM cNbpKx"
    />
    <div
      className="sc-fjdhpX kSIlTd"
    />
    <div
      className="sc-jzJRlG jCGiTt"
    />
    <div
      className="sc-cSHVUG hgwnda"
    />
  </div>
  <div
    className="sc-bxivhb hGBFui"
  />
  <div
    className="sc-EHOje eCkAKG"
  >
    <a
      className="sc-gZMcBi cQWaow"
      href={undefined}
      rel="noopener noreferrer"
      target="_blank"
      title="View my LinkedIn Profile"
    />
    <a
      className="sc-iwsKbI cbXTBy"
      href={undefined}
      rel="noopener noreferrer"
      target="_blank"
      title="View my GitHub Profile"
    />
  </div>
  <div
    className="sc-htoDjs jLBQXO"
  >
    <div
      className="sc-bZQynM gZGNLj"
    >
      <div
        className="sc-gzVnrw fOlWni"
      >
        <h3
          className="sc-bwzfXH GPZWx"
        >
          Front End Web Developer
        </h3>
      </div>
      <h1
        className="sc-bdVaJa dJVMHP"
      >
        jack coventry
      </h1>
    </div>
  </div>
</div>
`;
